### PR TITLE
Update green horizon and ball detectors to use multiclass visual mesh

### PR DESCRIPTION
**Pull request by tayloryoung6396**

### CODE REVIEW

**Comment by tayloryoung6396** [comment]
LGTM

File: module/vision/BallDetector/data/config/BallDetector.yaml [hunk 1]
```diff
@@ -1,48 +1,7 @@
...
-    # random_subset             a subset of the training samples (random)
-    # static_spread             a maximally spread subset of training samples (repeatable)
-    # random_spread             a maximally spread subset of training samples (random start)
+confidence_threshold: 0.7
+cluster_points: 3
+minimum_ball_distance: 0.20
+distance_disagreement: 0.30
+maximum_deviation: 0.15
+ball_angular_cov: [0.1, 0.01, 0.01]
+debug: false
```
Comment (MattAmos) [inline module/vision/BallDetector/data/config/BallDetector.yaml]: Maybe add comments to describe what each parameter does

Comment (MattAmos) [inline module/vision/BallDetector/data/config/BallDetector.yaml]: Maybe add comments to define what each param does

File: module/vision/GreenHorizonDetector/data/config/GreenHorizonDetector.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+confidence_threshold: 0.7
+cluster_points: 5
+fudge_factor: 1.0
+debug: false
```
Comment (MattAmos) [inline module/vision/GreenHorizonDetector/data/config/GreenHorizonDetector.yaml]: Add param comments (especially `fudge_factor`?)

Comment (MattAmos) [reply module/vision/GreenHorizonDetector/data/config/GreenHorizonDetector.yaml]: Remove `fudge_factor` if not used (isn't being read into GreenHorizonDetector)

File: module/vision/GreenHorizonDetector/src/GreenHorizonDetector.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    public:
+        /// @brief Called by the powerplant to build and setup the GreenHorizonDetector reactor.
+        explicit GreenHorizonDetector(std::unique_ptr<NUClear::Environment> environment);
+
+    private:
+        struct {
+            float confidence_threshold;
+            uint cluster_points;
+            float fudge_factor;
```
Comment (MattAmos) [inline module/vision/GreenHorizonDetector/src/GreenHorizonDetector.h]: Remove `fudge_factor` if not used

File: module/vision/GreenHorizonDetector/data/config/GreenHorizonDetector.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 confidence_threshold: 0.7       # How confident do we need to be that this is a green horizon point
 cluster_points: 5               # How many points do we need to claim this cluster is a viable part of the green horizon
-fudge_factor: 1.0               # Push the green horizon radially outwards by this distance (in m)
+distance_offsett: 1.0           # Push the green horizon radially outwards by this distance (in m)
```
Comment (TrentHouliston) [inline module/vision/GreenHorizonDetector/data/config/GreenHorizonDetector.yaml]: That sure is a lot of ts...

**Comment by TrentHouliston** [comment]
Change the title of this PR, young/green_horizon describes almost 0 of what’s in this pr

**Comment by Bidski** [comment]
Screenshot from 2019-06-30 14-41-20

**Pull request merged**